assembly/src/library.rs

```diff
@@ -118,6 +118,23 @@ mod use_std {
             })
         }
 
+        /// Read a library from a file.
+        pub fn read_from_file<P>(path: P) -> Result<MaslLibrary, LibraryError>
+        where
+            P: AsRef<Path>,
+        {
+            // convert path to str
+            let path_str = path.as_ref().to_str().unwrap_or("path contains invalid unicode");
```

Comment:
Question: should this return an error rather than panicking? Or this should never happen?

Reply:
`to_str` returns an `Option<&str>`.  It returns `None` if the `path` is a valid path that contains invalid unicode characters.  In this case we set `path_str = path contains invalid unicode` such that if we display errors later on this is communicated to the user.  It will never panic.

Reply:
Ah! Makes sense.